Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -19,24 +19,78 @@ module.exports.Quantity = class Quantity extends Expression
+      @value >= parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration >= otherSmallestDuration
+    else null
 
   after: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value > parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration > otherSmallestDuration
+    else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value < parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
     if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
 
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+# Hash of time units and their UCUM equivalents
+# See http://unitsofmeasure.org/ucum.html#para-31
+time_units = {'years': 'year', 'year': 'year', 'a': 'year', 'ANN': 'year', 'ann': 'year'
+  , 'months': 'month', 'month':'month', 'mo': 'month', 'MO': 'month'
+  , 'weeks': 'week', 'week': 'week', 'wk', 'week'
+  , 'days': 'day', 'day':'day', 'd': 'day', 'D': 'day'
+  , 'hours': 'hour', 'hour': 'hour', 'h': 'hour', 'H': 'hour'
+  , 'minutes': 'minute', 'minute': 'minute', 'min': 'minute', 'MIN': 'minute'
+  , 'seconds':'second', 'second':'second', 's': 'second'
+  , 'milliseconds' : 'millisecond', 'millisecond' : 'millisecond', 'ms': 'millisecond'
+  }
 
 clean_unit = (units) ->
-  if time_units[units] then time_units[units] else units
+  if time_units[units]
+    time_units[units]
+  else units
+
+# The smallest common duration is the millisecond
+smallestDuration = (qty) ->
+  if parseFloat qty.value
+    millivalue = switch
+      when clean_unit(qty.unit) == 'second' then qty.value * 1000
+      when clean_unit(qty.unit) == 'minute' then qty.value * 60 * 1000
+      when clean_unit(qty.unit) == 'hour' then qty.value * 60 * 60 * 1000
+      when clean_unit(qty.unit) == 'day' then qty.value * 24 * 60 * 60 * 1000
+      when clean_unit(qty.unit) == 'week' then qty.value * 7 * 24 * 60 * 60 * 1000
+      # Only supporting year durations based on a Julian year
+      # See http://unitsofmeasure.org/ucum.html#para-31
```

Comment:
i think we decided to change this to gregorian based on the emails that just occurred